src/components/WithScroll/index.js

```diff
@@ -0,0 +1,59 @@
+import React, { Component } from "react";
+
+const WithScroll = ComposedComponent =>
+  class ScrollDecorator extends Component {
+    constructor() {
+      super();
+
+      this.state = {
```

Comment:
you don't need defined `this.state` in `constructor`

`constructor` in new react is useless